File: 1/1-15.hs
-- Define the cube function
cube :: Double -> Double
cube x = x * x * x

-- Define the p function
p :: Double -> Double
p x = 3 * x - 4 * cube x

-- Define the sine function
sine :: Double -> Double
sine angle
    | abs angle <= 0.1 = angle
    | otherwise = p (sine (angle / 3.0))

-- Evaluate sine 12.15
result = sine 12.15

File: 1/1-15.rs
// Define the cube function
fn cube(x: f64) -> f64 {
    x * x * x
}

// Define the p function
fn p(x: f64) -> f64 {
    3.0 * x - 4.0 * cube(x)
}

// Define the sine function
fn sine(angle: f64) -> f64 {
    if angle.abs() <= 0.1 {
        angle
    } else {
        p(sine(angle / 3.0))
    }
}

// Evaluate sine(12.15)
let result = sine(12.15);